pkg/collect/host_subnetavailable.go

```diff
@@ -0,0 +1,287 @@
+package collect
+
+import (
+	"bytes"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/replicatedhq/troubleshoot/pkg/debug"
+)
+
+type SubnetStatus string
+
+const (
+	SubnetStatusAvailable     = "a-subnet-is-available"
+	SubnetStatusNoneAvailable = "no-subnet-available"
+)
+
+type SubnetAvailableResult struct {
+	CIDRRangeAlloc string `json:"CIDRRangeAlloc"`
+	DesiredCIDR    int    `json:"desiredCIDR"`
+	// If subnet-available, at least 1 of the DesiredCIDR size is available within CIDRRangeAlloc
+	Status SubnetStatus `json:"status"`
+}
+
+type CollectHostSubnetAvailable struct {
+	hostCollector *troubleshootv1beta2.SubnetAvailable
+	BundlePath    string
+}
+
+func (c *CollectHostSubnetAvailable) Title() string {
+	return hostCollectorTitleOrDefault(c.hostCollector.HostCollectorMeta, "Subnet Available")
+}
+
+func (c *CollectHostSubnetAvailable) IsExcluded() (bool, error) {
+	return isExcluded(c.hostCollector.Exclude)
+}
+
+func (c *CollectHostSubnetAvailable) Collect(progressChan chan<- interface{}) (map[string][]byte, error) {
+	procNetRoute, err := os.ReadFile("/proc/net/route")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read contents of /proc/net/route")
+	}
+
+	routes, err := parseProcNetRoute(string(procNetRoute))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse /proc/net/route")
+	}
+	debug.Printf("Routes: %+v\n", routes)
+
+	// IPv4 only right now...
+	if c.hostCollector.DesiredCIDR < 1 || c.hostCollector.DesiredCIDR > 32 {
+		return nil, errors.Wrap(err, fmt.Sprintf("CIDR range size %d invalid, must be between 1 and 32", c.hostCollector.DesiredCIDR))
+	}
+
+	splitCIDRRangeAlloc := strings.Split(c.hostCollector.CIDRRangeAlloc, "/")
+	if len(splitCIDRRangeAlloc) != 2 {
+		return nil, errors.Wrap(err, fmt.Sprintf("CIDRRangeAlloc value %s invalid, expected format x.x.x.x/##", c.hostCollector.CIDRRangeAlloc))
+	}
+	maskInt, err := strconv.Atoi(splitCIDRRangeAlloc[1])
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("CIDRRangeAlloc mask %s invalid, expected integer", splitCIDRRangeAlloc[1]))
+	}
+	if maskInt < 1 || maskInt > 32 {
```

Comment:
`/0` is valid

Reply:
I agree it's probably safe/valid to allow `/0` here, basically choose X subnet size from "the entirety of IPv4 available address space"

Reply:
https://github.com/replicatedhq/troubleshoot/pull/1004/commits/d461aa9d7f5e18702a950bb862bb3da7696be36b